class-sc-wp-updates-notifier.php

```diff
@@ -659,18 +697,127 @@ public function sc_wpun_settings_validate( $input ) {
 				add_settings_error( 'sc_wpun_settings_main_hide_updates', 'sc_wpun_settings_main_hide_updates_error', __( 'Invalid hide updates value entered', 'wp-updates-notifier' ), 'error' );
 			}
 
-			if ( isset( $_POST['submitwithemail'] ) ) {
-				add_filter( 'pre_set_transient_settings_errors', array( $this, 'send_test_email' ) );
+			// Validate email notification settings.
+			if ( ! empty( $input['notify_to'] ) ) {
+				$emails_to = explode( ',', $input['notify_to'] );
+				if ( $emails_to ) {
+					$sanitized_emails = array();
+					$was_error        = false;
+					foreach ( $emails_to as $email_to ) {
+						$address = sanitize_email( trim( $email_to ) );
+						if ( ! is_email( $address ) ) {
+							add_settings_error( 'sc_wpun_settings_email_notifications_notify_to', 'sc_wpun_settings_email_notifications_notify_to_error', __( 'One or more email to addresses are invalid', 'wp-updates-notifier' ), 'error' );
+							$was_error = true;
+							break;
+						}
+						$sanitized_emails[] = $address;
+					}
+					if ( ! $was_error ) {
+						$valid['notify_to'] = implode( ',', $sanitized_emails );
+					}
+				}
+			} else {
+				$valid['notify_to'] = '';
+			}
+
+			if ( ! empty( $input['notify_from'] ) ) {
+				$sanitized_email_from = sanitize_email( $input['notify_from'] );
+				if ( is_email( $sanitized_email_from ) ) {
+					$valid['notify_from'] = $sanitized_email_from;
+				} else {
+					add_settings_error( 'sc_wpun_settings_email_notifications_notify_from', 'sc_wpun_settings_email_notifications_notify_from_error', __( 'Invalid email from entered', 'wp-updates-notifier' ), 'error' );
+				}
+			} else {
+				$valid['notify_from'] = '';
+			}
+
+			$email_notifications = absint( isset( $input['email_notifications'] ) ? $input['email_notifications'] : 0 );
+			if ( $email_notifications > 1 ) {
```

Comment:
Do we need to Yoda here?

```suggestion
			if ( 1 < $email_notifications ) {
```